src/lexers/javascript-lexer.js

```diff
@@ -124,6 +124,7 @@ export default class JavascriptLexer extends BaseLexer {
     ) {
       const { typeArguments } = tFunctionParam.type
       if (
+        typeArguments &&
```

Comment:
we also need a fix in the lines below

```jsx
tFunctionParam &&
      tFunctionParam.type &&
      tFunctionParam.type.typeName &&
      tFunctionParam.type.typeName.text === 'TFunction')
```

Reply:
@sibelius Property `typeName` is not optional according to [type definitions](https://github.com/microsoft/TypeScript/blob/main/lib/typescript.d.ts#L946).

Reply:
type definitions can be wrong